devices/libsdio/README.md

```diff
@@ -0,0 +1,277 @@
+
+---
+
+```c
+int sdio_transferBulk(sdio_dir_t dir, int blockMode, uint32_t address, uint8_t area, uint8_t *data, size_t len);
+```
+
+This function initiates an indirect, multi-byte transfer of up to 2048 bytes at once. As is the case with `sdio_transferDirect`, this call can service bi-directional transfers.
+
+Parameters:
+
+| parameter                | description                             |
+|--------------------------|-----------------------------------------|
+| [in] `sdio_dir_t dir`    | transfer direction                      |
+| [in] `int blockMode`     | wheter to use blocks for transfer (1/0) |
+| [in] `uint32_t address`  | card base address to access             |
+| [in] `uint8_t area`      | card I/O area index to access           |
+| [in/out] `uint8_t *data` | bi-directional multi byte buffer        |
+| [in] `size_t len`        | total transfer size in bytes            |
+
+> **NOTE:** `address` parameter specifies only the base address of the transfer. If `blockMode` is set to 1, then the address is automatically incremented by the card with every following block.
+
+Returns:
+
+| code        | description                                                 |
+|-------------|-------------------------------------------------------------|
+|  0          | success                                                     |
+| -1          | failure                                                     |
+| EINVAL (22) | `blocksz` is unset or `len` is not a multiple of block size |
+
+> **NOTE:** Every failure is accompanied by explicit fault description printed into standard output.
+
+---
+
+```c
+void sdio_eventRegister(uint8_t event, sdio_event_handler_t handler, void *arg);
+```
+
+This function registers an event handler to be called when a given interrupt event occurs. An interrupt event will not be signalled until its detection is enabled by `sdio_eventEnable`. Please note that only one handler can be registered per event - calling this function multiple times for the same event will result in the previous handler being overwritten.
+
+Parameters:
+
+| parameter                           | description                             |
+|-------------------------------------|-----------------------------------------|
+| [in] `uint8_t event`                | event to be handled                     |
+| [in] `sdio_event_handler_t handler` | pointer to event handler function       |
+| [in/out] `void *arg`                | argument for the handler function       |
+
+> **NOTE:** `event` argument is passed using appropriate define beginning with `SDIO_EVENT`.
+
+Returns:
+
+- nothing
+
+---
+
+```c
+void sdio_eventEnable(uint8_t event, int enabled);
+```
+
+This function can enable or disable interrupt event signalling of the SD host controller. If a handler is registered via `sdio_eventRegister` for a given enabled event, it will be called when an interrupt fires.
```

Comment:
Sooo, I think it shouldn't be called eventEnable then (if it also disables). Maybe eventControl or eventSet or eventSetState or something?
```suggestion
This function enables or disables an interrupt event signalling of the SD host controller. If a handler is registered via `sdio_eventRegister` for a given enabled event, it will be called when an interrupt appears.
```